File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> set = new HashSet<>();
        for(int i : nums)
        {
            set.add(i);
        }
        int max =0;
        for(int i : set)
        {
            if(!set.contains(i-1))
            {
                int start = i;
                int count=1;
                while(set.contains(start +1))
                {
                    start++;
                    count++;
                }
                max = Math.max(count,max);
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-4.go
func hasDuplicate(nums []int) bool {
    hashset := make(map[int]struct{})
    for i:= 0; i< len(nums);i++{
        _,b := hashset[nums[i]]
        if b {
            return true
        } 
        hashset[nums[i]] = struct{}{}
    }
    return false
}
